src/views/Dashboard/Customer/Home/MyTest/TestResult.vue

```diff
@@ -81,45 +88,232 @@
-        fileTitle: "Download Raw Data",
-        fileSubTitle: "Download Your Genomic Data"
-      }
-    ],
+    files: [],
+    dataType: "report",
+    fileDownloadIndex: 0,
+    baseUrl: "https://ipfs.io/ipfs/",
     dummyResult: {
       title: "Whole Genome Sequencing Test Report",
       subTitle: "GSI Lab, 5 July 2021",
       content: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
     }
   }),
 
+  async mounted() {
+    this.resultLoading = true;
+    this.specimentNumberInput = this.$route.params.dna_sample_tracking_id;
+    this.privateKey = hexToU8a(this.mnemonicData.privateKey);
+    this.ownerAddress = this.wallet.address;
+    await this.getSpciments();
+    await this.getLabServices();
+    await this.getFileUploaded();
+    await this.decryptWallet();
+  },
+
   methods: {
+    async getSpciments() {
+      try {
+        console.log(this.speciment, this.specimentNumberInput, this.api);
+        this.speciment = await queryDnaTestResults(
+          this.api,
+          this.specimentNumberInput
+        );
+      } catch (error) {
+        this.resultLoading = false;
+        console.log(error);
+      }
+    },
+
+    async getLabServices() {
+      try {
+        this.lab = await queryLabsById(this.api, this.speciment.lab_id);
+        this.order = await getOrdersDetail(this.api, this.speciment.order_id);
+        this.ownerAddress = this.order.customer_eth_address;
+        this.services = await queryServicesById(this.api, this.order.service_id);
+
+        this.serviceCategory = this.services.info.category;
+        this.serviceName = this.services.info.name;
+      } catch (error) {
+        this.resultLoading = false;
+        this.services = [];
+      }
+    },
+
+    async getFileUploaded() {
+      try {
+        if (this.speciment.report_link !== "") {
+          this.files.push({
+            fileType: "report",
+            fileName: this.serviceName + " Report",
+            fileLink: this.speciment.report_link,
+            fileTitle: "Download Report",
+            fileSubTitle: "Download Your Test Report"
+          });
+        }
+
+        if (this.speciment.result_link !== "") {
+          this.files.push({
+            fileType: "result",
+            fileName: this.serviceName + " Result",
+            fileLink: this.speciment.result_link,
+            fileTitle: "Download Raw Data",
+            fileSubTitle: "Download Your Genomic Data"
+          });
+        }
+      } catch (error) {
+        this.resultLoading = false;
+      }
+    },
+
+    async decryptWallet() {
```

Comment:
You're using `decryptWallet` a whole lot, and the functionality is to download from IPFS? Can you rename this method, because I think the method name doesn't fit at all to the functionality IMO?